### PR TITLE
Replace usage of Whitelist with Allowlist within Kubelet's sysctl package

### PR DESCRIPTION
**Pull request by wesleyw72**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
There has been a recommendation ([link](https://github.com/kubernetes/community/blob/master/wg-naming/recommendations/002-blacklist-whitelist-allowlist-denylist.md)) from wg-naming to replace uses of Whitelist with Allowlist. This PR makes that change for the sysctl package within the kubelet.
 
#### Which issue(s) this PR fixes:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wesleyw72** [comment]
/priority/important-longterm

**Comment by ehashman** [comment]
Found a few more that you missed:

```
ehashman@fedora:~/src/k8s/pkg/kubelet$ git grep -i white
apis/config/types.go:   // A comma separated whitelist of unsafe sysctls or sysctl patterns (ending in *).
cm/cgroup_manager_linux.go:     whitelistControllers := sets.NewString("cpu", "cpuacct", "cpuset", "memory", "systemd", "pids")
cm/cgroup_manager_linux.go:             whitelistControllers.Insert("hugetlb")
cm/cgroup_manager_linux.go:             if !whitelistControllers.Has(controller) {
kubelet.go:     sysctlwhitelist "k8s.io/kubernetes/pkg/security/podsecuritypolicy/sysctl"
kubelet.go:     // Safe, whitelisted sysctls can always be used as unsafe sysctls in the spec.
kubelet.go:     safeAndUnsafeSysctls := append(sysctlwhitelist.SafeSysctlWhitelist(), allowedUnsafeSysctls...)
kubelet_pods.go:        // TODO: whitelist logs we are willing to serve
sysctl/allowlist_test.go:               _, err := NewAllowlist(append(sysctl.SafeSysctlWhitelist(), test.sysctls...))
sysctl/allowlist_test.go:func TestWhitelist(t *testing.T) {
sysctl/allowlist_test.go:       w, err := NewAllowlist(append(sysctl.SafeSysctlWhitelist(), "kernel.msg*", "kernel.sem"))
```

**Comment by wesleyw72** [comment]
@ehashman Thanks, I was targeting just the sysctl package within the kubelet and was going to do the others in another PR. However, I'm happy to include those as part of this. Will get that done.

**Comment by wesleyw72** [comment]
@ehashman I think that's all addressed now.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -786,14 +786,14 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 	klet.evictionManager = evictionManager
 	klet.admitHandlers.AddPodAdmitHandler(evictionAdmitHandler)
 
-	// Safe, whitelisted sysctls can always be used as unsafe sysctls in the spec.
+	// Safe, allowlisted sysctls can always be used as unsafe sysctls in the spec.
```
Comment (endocrimes) [inline pkg/kubelet/kubelet.go:789]:
```suggestion
	// Safe, allowed sysctls can always be used as unsafe sysctls in the spec.
```

tiny inconsequential nit, but this reads a little better to me.

**Review by endocrimes — COMMENTED** [review]
image

/lgtm

**Review by endocrimes — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**